src/DashboardService.Business/Priority/GetPrioritiesCommand.cs

```diff
@@ -1,24 +1,38 @@
 using LT.DigitalOffice.DashboardService.Business.Priority.Interfaces;
 using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.DashboardService.Models.Db;
 using LT.DigitalOffice.DashboardService.Models.Dto.Models;
 using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority.Filters;
 using LT.DigitalOffice.Kernel.Responses;
-using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.DashboardService.Business.Priority;
 
 public class GetPrioritiesCommand : IGetPrioritiesCommand
 {
   private readonly IPriorityRepository _priorityRepository;
+  private readonly IPriorityInfoMapper _priorityInfoMapper;
 
-  public GetPrioritiesCommand(IPriorityRepository priorityRepository)
+  public GetPrioritiesCommand(
+    IPriorityRepository priorityRepository,
+    IPriorityInfoMapper priorityInfoMapper)
   {
     _priorityRepository = priorityRepository;
+    _priorityInfoMapper = priorityInfoMapper;
   }
   
-  public Task<FindResultResponse<PriorityInfo>> ExecuteAsync(GetPrioritiesFilter filter)
+  public async Task<FindResultResponse<PriorityInfo>> ExecuteAsync(GetPrioritiesFilter filter, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    (List<DbPriority> dbPriorities, int totalCount) result = await _priorityRepository.GetAllAsync(filter, ct);
+
+    return new()
+    {
+      Body = result.dbPriorities.Select(p => _priorityInfoMapper.Map(p)).ToList(), 
```

Comment:
Select().ToList() -> ConvertAll()

Reply:
Perhaps dbPriorities is ICollection and it does not contain ConvertAll

Reply:
++